### ISSUE TITLE
twisted.conch.test.test_cftp and twisted.conch.test.test_conch fail on Windows if PyCrypto and pyasn1 are installed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3916]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3916|
|Type|defect|
|Created|2009-07-16 15:31:02Z|
|Branch|https://github.com/twisted/twisted/tree/conch-test-skips-3916|

```
===============================================================================
[ERROR]: twisted.conch.test.test_cftp

Traceback (most recent call last):
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\trial\runner.py", line 563, in loadPackage
    module = modinfo.load()
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\python\modules.py", line 380, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\python\reflect.py", line 462, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\conch\test\test_cftp.py", line 37, in <module>
    from twisted.conch.test import test_ssh, test_conch
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\conch\test\test_conch.py", line 24, in <module>
    from twisted.conch.scripts.conch import SSHSession as StdioInteractingSession
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\conch\scripts\conch.py", line 18, in <module>
    import os, sys, getpass, struct, tty, fcntl, signal
  File "c:\python25\lib\tty.py", line 5, in <module>
    from termios import *
exceptions.ImportError: No module named termios
===============================================================================
[ERROR]: twisted.conch.test.test_conch

Traceback (most recent call last):
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\trial\runner.py", line 563, in loadPackage
    module = modinfo.load()
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\python\modules.py", line 380, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\python\reflect.py", line 462, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\conch\test\test_conch.py", line 24, in <module>
    from twisted.conch.scripts.conch import SSHSession as StdioInteractingSession
  File "c:\twistedbot2\WXP32-full2.5-scmikes-select\Twisted\twisted\conch\scripts\conch.py", line 18, in <module>
    import os, sys, getpass, struct, tty, fcntl, signal
  File "c:\python25\lib\tty.py", line 5, in <module>
    from termios import *
exceptions.ImportError: No module named termios
-------------------------------------------------------------------------------
```

<details><summary>Searchable metadata</summary>

```
trac-id__3916 3916
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_conch_test_skips_3916 branches/conch-test-skips-3916
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1247758262000000 1247758262000000
changetime__1286996480000000 1286996480000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27158]) Branching to 'conch-test-skips-3916'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

I was somewhat tempted to go into `test_ssh.py` with a hatchet, but I resisted.  I think the changes in the branch address the problem described in the ticket summary, and simplify the code a bit as well.  There's room for more improvement, but maybe that will happen later.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

OK, that can be helped, I guess. Please merge if the windows builders are green. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Well, greener.  http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/conch-test-skips-3916

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [27171]) Merge conch-test-skips-3916

Author: exarkun
Reviewer: therve
Fixes: [#3916](https://github.com/twisted/twisted/issues/3916)

Change two of Conch's test modules so that they skip when running on Windows
when both pyasn1 and PyCrypto are available.  Previously they would try to import
termios and fail, as this module is not available on Windows.

Also deleted is some left-over support code for Conch connection sharing, a feature
removed some time ago.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
